File: e25.js


function fib() {
  var start = Date.now();

  var f1 = 1,
    f2 = 1,
    temp = 0;


  for (var i = 3; i > 0; i++) {
    temp = f2;
    f2 = f2 + f1;
    f1 = temp;

    var str = f2.toString();
    var indexOfE = str.indexOf('e');
    if(indexOfE > -1) {

      var digits = str.substr(indexOfE + 2);
      console.log(i, digits);

      if (digits >= 999) {
        var end = Date.now();
        console.log(end - start);

        return i;

      }

    }
  }

}

console.log(fib());

File: e25.scala
def e25(): Int = {
  var a = BigInt("1")
  var b = BigInt("1")
  var index = 3

  while(index > 0) {
    val temp = a + b
    if (temp.toString().length() > 999) {
      return index
    }
    index += 1
    a = b
    b = temp
  }
  index
}

println(e25())